### PR TITLE
chore(documentation): Add api documentation

### PR DESCRIPTION
**Pull request by nelmandela**
### What does this PR do?

This PR adds documentation for the art-backend endpoints.

### Description of Task to be completed?

  •Install django-rest-swagger
  •Add a /doc path to the url file

How should this be manually tested?

  •Locally, navigate to` http://127.0.0.1:8000/docs/`
  •Test the endpoints available here

Any background context you want to provide?

The frontend developers require a doc to refer to. This addresses their program.

What are the relevant pivotal tracker stories?

[#157681976](https://www.pivotaltracker.com/story/show/157681976)

Screenshots (if appropriate)
screen shot 2018-05-21 at 18 46 06

### CODE REVIEW

File: art/urls.py [hunk 1]
```diff
@@ -18,19 +18,32 @@
...
+from drf_yasg import openapi
 
 from api import urls
 
-schema_view = get_swagger_view(title='ART API')
+schema_view = get_schema_view(
+    openapi.Info(
+        title="ART API",
+        default_version='v1',
+        description="Test description",
```
Comment (egichuri) [inline art/urls.py]: test description?

File: art/urls.py [hunk 2]
```diff
@@ -18,19 +18,32 @@
...
+        terms_of_service="https://www.google.com/policies/terms/",
+        contact=openapi.Contact(email="contact@art.local"),
```
Comment (egichuri) [inline art/urls.py]: i think we can remove  terms and contact

File: requirements.txt [hunk 3]
```diff
@@ -1,26 +1,67 @@
+astroid==1.6.1
```
Comment (egichuri) [inline requirements.txt]: Are all these dependencies part of your changes?

Comment (nelmandela) [reply requirements.txt]: On it

**Comment by egichuri** [comment]
@nelmandela also check https://stackoverflow.com/a/39554754 on how to enable token auth

File: art/urls.py [hunk 4]
```diff
@@ -18,19 +18,32 @@
...
     path('admin/', admin.site.urls),
     path('api/v1/', include(urls)),
     path('jet/', include('jet.urls', 'jet')),
     path('jet/dashboard/', include('jet.dashboard.urls', 'jet-dashboard')),
-    path('docs/', schema_view)
+    path('docs/', schema_view.with_ui('swagger',
+                                      cache_timeout=None),
+         name='schema-swagger-ui')
 ]
+
```
[the remark this answers is not in the document]
Comment (egichuri) [reply art/urls.py]: oh.. and set `public=True` ==> https://drf-yasg.readthedocs.io/en/stable/readme.html#quickstart

File: art/urls.py [hunk 5]
```diff
@@ -18,19 +18,32 @@
...
 from django.conf.urls.static import static
 from django.urls import path
-from rest_framework_swagger.views import get_swagger_view
+from drf_yasg.views import get_schema_view
+from drf_yasg import openapi
 
 from api import urls
 
-schema_view = get_swagger_view(title='ART API')
+schema_view = get_schema_view(
```
Comment (alexkiura) [inline art/urls.py]: Also let's be consistent with the string quotes. Noticed usage of both double and single quotes. Let's refactor the file to use single quotes.

**Comment by nelmandela** [comment]
having an issue pip installing  drf-yasg[validation] ,been unable to find a hack on how to go about it

File: settings/base.py [hunk 6]
```diff
@@ -190,3 +190,15 @@
 csv = Csv(cast=lambda s: tuple(s.split(':')))
 
 ADMINS = csv(os.environ.get('ADMINS', ''))
+
+SWAGGER_SETTINGS = {
+    'SECURITY_DEFINITIONS': {
+        'basic': {
+            'type': 'basic'
+        }
```
Comment (egichuri) [inline settings/base.py]: @nelmandela we need token authentication for this

File: art/urls.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (egichuri) [reply art/urls.py]: @nelmandela let's remove the docs stuff from this file since we've moved it to `api/urls.py`

File: settings/base.py (hunk 6 above)
Comment (nelmandela) [reply settings/base.py]: on it

File: art/urls.py (hunk 5 above)
Comment (nelmandela) [reply art/urls.py]: @alexkiura done

**Pull request merged**